balrogscript/src/balrogscript/submitter/cli.py

```diff
@@ -154,12 +154,17 @@ def run(
 
         if self.backend_version == 2:
             log.info("Using backend version 2...")
-            blob["schema_version"] = self.schemaVersion
-            blob["hashFunction"] = hashFunction
-            blob["name"] = name
-            data = {"product": productName, "blob": blob}
             url = self.api_root + "/v2/releases/" + name
+            data = {"product": productName, "blob": {}}
             session = get_balrog_session(auth0_secrets=self.auth0_secrets)
+            current_data = balrog_request(session, "get", url)
+            if current_data:
+                data["blob"] = current_data["blob"]
+                data["old_data_versions"] = current_data["data_versions"]
+            data["blob"].update(blob)
```

Comment:
This causes race conditions if toplevel and secondary toplevel submission run at the same time - we end up losing some of the fileUrls. If we deep merge here (so we keep all entries) it should work fine.

Reply:
https://stackoverflow.com/questions/7204805/how-to-merge-dictionaries-of-dictionaries/7205107#7205107

https://www.electricmonk.nl/log/2017/05/07/merging-two-python-dictionaries-by-deep-updating/

https://y.tsutsumi.io/2017/04/24/deepmerge-deep-merge-dictionaries-lists-and-more-in-python/